parsers/jats/PublicationParser.inc.php

```diff
@@ -121,12 +121,144 @@ public function getPublication(): \Publication
 		// Handle PDF galley
 		$this->_insertPDFGalley($publication);
 
+		// Record this XML itself
+		$this->_insertXMLSubmissionFile($publication);
+
 		// Publishes the article
 		\Services::get('publication')->publish($publication);
 
 		return $publication;
 	}
-	
+		
+	/**
+	 * Inserts the PDF galley
+	 * @param \Publication $publication
+	 */
+	private function _insertXMLSubmissionFile(\Publication $publication): void
+	{
+		import('lib.pkp.classes.file.SubmissionFileManager');
+		$splfile = $this->getArticleEntry()->getMetadataFile();
+		$filename = $splfile->getPathname();
+
+		$genreId = \GENRE_CATEGORY_DOCUMENT;
+		$fileSize = filesize($filename);
+		$fileType = "text/xml";
+		$fileStage = \SUBMISSION_FILE_PRODUCTION_READY;
+		$userId = $this->getConfiguration()->getUser()->getId();
+
+		$submission = $this->getSubmission();
+
+		$submissionFileDao = \DAORegistry::getDAO('SubmissionFileDAO');
+		$newSubmissionFile = $submissionFileDao->newDataObjectByGenreId($genreId);
+		$newSubmissionFile->setSubmissionId($submission->getId());
+		$newSubmissionFile->setSubmissionLocale($submission->getLocale());
+		$newSubmissionFile->setGenreId($genreId);
+		$newSubmissionFile->setFileStage($fileStage);
+		$newSubmissionFile->setDateUploaded(\Core::getCurrentDate());
+		$newSubmissionFile->setDateModified(\Core::getCurrentDate());
+		$newSubmissionFile->setOriginalFileName($splfile->getFilename());
+		$newSubmissionFile->setUploaderUserId($userId);
+		$newSubmissionFile->setFileSize($fileSize);
+		$newSubmissionFile->setFileType($fileType);
+		$newSubmissionFile->setName($splfile->getFilename(), $submission->getLocale());
+
+		$insertedSubmissionFile = $submissionFileDao->insertObject($newSubmissionFile, $filename);
+
+		foreach ($this->select('//asset') as $asset) {
```

Comment:
Changed... kinda.  Still targeting the node: 1) because we might want other data from the node in the future, 2) because namespacing works by default for attribute selection, but not for the xpath (need to register `xlink`).